docs/website/content/v0.3/en/configuration/v1alpha1.md

```diff
@@ -0,0 +1,276 @@
+```
+
+nolint: lll
+
+## Reference
+
+### type
+
+Type: `string`
+
+Valid Values:
+
+- ``init``
+- ``controlplane``
+- ``join``
+
+The machine's `type` defines the role of the machine within the cluster.
+
+#### Init
+
+Init node type designates the first control plane node to come up.
+You can think of it like a bootstrap node.
+This node will perform the initial steps to bootstrap the cluster -- generation of TLS assets, starting of the control plane, etc.
+
+#### Control Plane
+
+Control Plane node type designates the node as a control plane member.
+This means it will host etcd along with the Kubernetes master components such as API Server, Controller Manager, Scheduler.
+
+#### Worker
+
+Worker node type designates the node as a worker node.
+This means it will be an available compute node for scheduling workloads.
+
+### token
+
+Type: `string`
+
+The `token` is used by a machine to join the PKI of the cluster.
+Using this token, a machine will create a certificate signing request (CSR), and request a certificate that will be used as its' identity.
+
+Examples:
+
+```yaml
+token: 328hom.uqjzh6jnn2eie9oi
+```
+
+> Warning: It is important to ensure that this token is correct sinc a machine's certificate has a short TTL by default
+
+### ca
+
+Type: `PEMEncodedCertificateAndKey`
```

Comment:
the type here is technically correct but not useful to end users. i guess we could try to get fancy and link to godoc here to articulate the type definition, but not sure i care for that either.

Reply:
Another limitation of AST. I intend on linking to the local sections that contains the info about the type, but things get weird if the type is from another package. We may have to live with this until we can spend a lot more time digging into the AST library. At the very least, we now have sections dedicated to non-builtin-in types that are part of the current package.